agent/engine/docker_task_engine.go

```diff
@@ -657,6 +663,35 @@ func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Con
 	return engine.serialPull(task, container)
 }
 
+// useLocalImageCache returns true if local image cache should be used, or return false to continue
+// pulling image, by inspecting the agent pull behavior variable defined in config. The caller has
+// to make sure the container passed in is not an internal container.
+func (engine *DockerTaskEngine) useLocalImageCache(agentPullBehavior config.AgentPullBehaviorType,
+	container *api.Container,
+	taskArn string) bool {
+	// If the agent pull behavior is never, we don't try to pull the image,
+	// try to use local image cache instead.
+	if agentPullBehavior == config.NeverAgentPullBehavior {
+		seelog.Infof("Task engine [%s]: use cached image directly for container %s",
+			taskArn, container.Name)
+		// It's not an internal container, so it's safe to add container reference.
+		engine.updateContainerReference(false, container, taskArn)
+		return true
+	} else if agentPullBehavior == config.OnceAgentPullBehavior {
+		// If this image has been pulled successfully before, don't pull the image,
+		// otherwise pull the image as usual, regardless whether the image exists or not
+		// (the image can be prepopulated with the AMI and never be pulled).
+		imageState := engine.imageManager.GetImageStateFromImageName(container.Image)
+		if imageState != nil && imageState.PullSucceeded {
+			seelog.Infof("Task engine [%s]: image %s has been pulled, use it directly for container %s",
```

Comment:
i'd suggest changing this slightly to :`image %s has been pulled once, not pulling it again`

Reply:
OK